src/main/java/org/jabref/gui/shared/SharedDatabaseLoginDialogView.java

```diff
@@ -0,0 +1,320 @@
+import java.nio.file.Paths;
+import java.security.GeneralSecurityException;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import javafx.event.ActionEvent;
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.control.ButtonBar.ButtonData;
+import javafx.scene.control.ButtonType;
+import javafx.scene.control.CheckBox;
+import javafx.scene.control.ComboBox;
+import javafx.scene.control.PasswordField;
+import javafx.scene.control.TextField;
+import javafx.scene.control.Alert.AlertType;
+
+import org.jabref.Globals;
+import org.jabref.JabRefException;
+import org.jabref.gui.BasePanel;
+import org.jabref.gui.DialogService;
+import org.jabref.gui.JabRefFrame;
+import org.jabref.gui.exporter.SaveDatabaseAction;
+import org.jabref.gui.help.HelpAction;
+import org.jabref.gui.util.BaseDialog;
+import org.jabref.gui.util.FileDialogConfiguration;
+import org.jabref.logic.help.HelpFile;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.logic.shared.DBMSConnectionProperties;
+import org.jabref.logic.shared.exception.InvalidDBMSConnectionPropertiesException;
+import org.jabref.logic.shared.prefs.SharedDatabasePreferences;
+import org.jabref.logic.shared.security.Password;
+import org.jabref.logic.util.FileType;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.model.database.shared.DBMSType;
+import org.jabref.model.database.shared.DatabaseLocation;
+import org.jabref.model.database.shared.DatabaseNotSupportedException;
+import org.jabref.preferences.JabRefPreferences;
+
+import com.airhacks.afterburner.views.ViewLoader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SharedDatabaseLoginDialogView extends BaseDialog<Void> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SharedDatabaseLoginDialogView.class);
+
+    @FXML private ComboBox<DBMSType> cmbDbType;
+    @FXML private TextField tbHost;
+    @FXML private TextField tbDb;
+    @FXML private TextField tbPort;
+    @FXML private TextField tbUser;
+    @FXML private PasswordField tbPwd;
+    @FXML private CheckBox chkRememberPassword;
+    @FXML private TextField tbFolder;
+    @FXML private Button btnBrowse;
+    @FXML private CheckBox chkAutosave;
+
+    @Inject private DialogService dialogService;
+
+    private final SharedDatabasePreferences prefs = new SharedDatabasePreferences();
+
+    private final ButtonType connect = new ButtonType("Connect", ButtonData.YES);
+    private final Button btnConnect;
+    private final JabRefFrame frame;
+
+    private DBMSConnectionProperties connectionProperties;
+    private SharedDatabaseLoginDialogViewModel viewModel;
+
+    public SharedDatabaseLoginDialogView(JabRefFrame frame) {
+        this.frame = frame;
+        this.setTitle(Localization.lang("Connect to shared database"));
+
+        this.getDialogPane().getButtonTypes().addAll(connect, ButtonType.CANCEL);
```

Comment:
It is preferable to set the buttons in the fxml
https://github.com/JabRef/jabref/blob/390c38740997201c6ab6c5e66bf53c2991507a48/src/main/java/org/jabref/gui/help/AboutDialog.fxml#L99-L100
and then add the action in the code-behind
https://github.com/JabRef/jabref/blob/390c38740997201c6ab6c5e66bf53c2991507a48/src/main/java/org/jabref/gui/help/AboutDialogView.java#L36

Reply:
Unfortunately this results in an NPE which I don't understand.  The lookUpButton returns null.

Reply:
Did you also tried it without the surrounding `buttontypes` tag?